msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/MultipleAccountPublicClientTests.java

```diff
@@ -0,0 +1,539 @@
+        MockHttpClient.reset();
+        MockHttpClient.setHttpResponse(
+                MockServerResponse.getMockCloudDiscoveryResponse(),
+                HttpClient.HttpMethod.GET,
+                MockTokenCreator.CLOUD_DISCOVERY_ENDPOINT_REGEX);
+        for (UserAccountData userData: testCaseData.userAccountsData) {
+            HttpResponse mockResponseForHomeTenant = MockServerResponse.getMockTokenSuccessResponse(
+                    userData.localAccountId,
+                    userData.tenantId,
+                    userData.cloud,
+                    testCaseData.getRawClientInfo(),
+                    userData.getFakeAccessToken());
+            MockHttpClient.setHttpResponse(mockResponseForHomeTenant,
+                    HttpClient.HttpMethod.POST,
+                    userData.cloud + "/.*");
+            performInteractiveAcquireTokenCall(testCaseData.userName, userData.cloud + "/organizations");
+            if(testCaseData.homeAccountId.equals(userData.localAccountId + "." + userData.tenantId)){
+                homeAccountSignedIn = true;
+            }
+        }
+    }
+
+    @Override
+    public String[] getScopes() {
+        return USER_READ_SCOPE;
+    }
+
+    @Override
+    public String getAuthority() {
+        return mApplication.getConfiguration().getDefaultAuthority().getAuthorityURL().toString();
+    }
+
+    @Override
+    public String getConfigFilePath() {
+        return TestConstants.Configurations.MULTIPLE_ACCOUNT_MODE_AAD_CONFIG_FILE_PATH;
+    }
+
+    private static class TestCaseData {
+        public final String userName;
+        public final String homeAccountId;
+        public final ArrayList<UserAccountData> userAccountsData;
+
+        public TestCaseData(
+                final String userName, String homeAccountId, ArrayList<UserAccountData> userAccountsData) {
+            this.userName = userName;
+            this.homeAccountId = homeAccountId;
+            this.userAccountsData = userAccountsData;
+        }
+        public String getRawClientInfo() {
+            final String uid = homeAccountId.split("\\.")[0];
+            final String utid = homeAccountId.split("\\.")[1];
+            final String claims = "{\"uid\":\"" + uid + "\",\"utid\":\"" + utid + "\"}";
+
+            return new String(Base64.encode(claims.getBytes(
+                    Charset.forName("UTF-8")), Base64.NO_PADDING | Base64.NO_WRAP | Base64.URL_SAFE));
+        }
+
+    }
+
+    private static class UserAccountData {
+        public final String cloud;
+        public final String localAccountId;
+
+        public final String tenantId;
+
+        public UserAccountData(final String cloud, final String localAccountId, final String tenantId){
+            this.cloud = cloud;
+            this.localAccountId = localAccountId;
+            this.tenantId = tenantId;
+        }
+        public String getFakeAccessToken() {
+            return "access_Token:" + cloud + "/" + tenantId + "/" + localAccountId;
```

Comment:
Is there a specific reason for this fake AT to be in this format? Is there some code internally that relies on this AT being in this format? If not, can we just return garbage here? How about just returning a random UUID string?

Reply:
but are we cracking open the AT to perform these assertions? If we are then that seems wrong and what is inside a real AT are the claims and it won't be formatted in the way as we have hard coded here

Reply:
no we are not cracking open, We are just verifying that the AT value is same before and after, so it can be anything I just kept it this way